### PR TITLE
Feat: Support Graphql subscription via websocket[INS-4390]

### LINKED ISSUE #6055: Support GraphQL subscriptions via WebSocket
### Expected Behavior

I want to be able to connect to GraphQL subscriptions via insomnia.

### Actual Behavior

I get 403 Forbidden errors

### Reproduction Steps

_No response_

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

As far as I am aware, javascript clients use the apollo graphql client to send and receive over a websocket (wss://url.com/graphql)

connection.

This would be amazing to support for insonmia as there is a gap in testing from the client for this feature.

### Insomnia Version

2023.2.2

### What operating system are you using?

Windows

### Operating System Version

Windows 11

### Installation method

Download from official website

### Last Known Working Insomnia version

_No response_

### PR DESCRIPTION
**Pull request by cwangsmv**
**Changes**:

- Using websocket connection for Graphql subscritpion operations

https://github.com/user-attachments/assets/97588083-118d-4678-9072-f6addbc23881

- Add logic to distinguish websocket subscription operation( Use websocket connection rather than http post request)
- Add handler to parse graphql websocket server messages. The protocol is defined in https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md

Closes https://github.com/Kong/insomnia/issues/6055

### CODE REVIEW

**Review by jackkav — COMMENTED** [review]
Curious the see how persistent the connection is. What if you navigate to another workspace and back? Or open a second instance of insomnia.

File: packages/insomnia/src/ui/components/request-url-bar.tsx [hunk 1]
```diff
@@ -213,11 +214,12 @@ export const RequestUrlBar = forwardRef<RequestUrlBarHandle, Props>(({
...
   });
 
-  const buttonText = isEventStreamRequest(activeRequest) ? 'Connect' : (downloadPath ? 'Download' : 'Send');
-  const borderRadius = isEventStreamRequest(activeRequest) ? 'rounded-sm' : 'rounded-l-sm';
+  const buttonText = isRealtimeRequest ? 'Connect' : (downloadPath ? 'Download' : 'Send');
+  const borderRadius = isRealtimeRequest ? 'rounded-sm' : 'rounded-l-sm';
   const { url, method } = activeRequest;
   const isEventStreamOpen = useReadyState({ requestId: activeRequest._id, protocol: 'curl' });
-  const isCancellable = currentInterval || currentTimeout || isEventStreamOpen;
+  const isGraphQLSubscriptionOpen = useReadyState({ requestId: activeRequest._id, protocol: 'webSocket' });
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/request-url-bar.tsx:221]: I'm a bit tempted to try to merge the action-bar.tsx into this one as its mostly copypasta and this is the only important line in that component.

**Review by jackkav — COMMENTED** [review]
really elegant implementation

**Comment by cwangsmv** [comment]
> Curious the see how persistent the connection is. What if you navigate to another workspace and back? Or open a second instance of insomnia.

It's a reuse of the current websocket connection logic so it will have the same behavior. Current when nav to another workspace and back, the connection will be closed

File: packages/insomnia/src/ui/components/request-url-bar.tsx (hunk 1 above)
Comment (cwangsmv) [reply packages/insomnia/src/ui/components/request-url-bar.tsx:221]: Agree, so both normal request and websocket request could use request-url-bar.tsx

Comment (jackkav) [reply packages/insomnia/src/ui/components/request-url-bar.tsx:221]: Can scope it out of this PR though.

File: packages/insomnia/package.json [hunk 2]
```diff
@@ -79,7 +79,9 @@
     "tough-cookie": "^4.1.3",
     "uuid": "^9.0.1",
     "yaml": "^1.6.0",
-    "yaml-source-map": "^2.1.1"
+    "yaml-source-map": "^2.1.1",
+    "graphql-ws": "^5.16.0",
+    "graphql": "^16.8.1"
```
Comment (jackkav) [inline packages/insomnia/package.json]: running `npm i -w insomnia -S graphql-ws` will fix this ordering issue

**Review by jackkav — DISMISSED** [review]

**Review by notjaywu — APPROVED** [review]
lgtm

**Pull request merged**